cmake/src.cmake

```diff
@@ -92,10 +106,23 @@ set(public_headers
   )
 set_target_properties(matio PROPERTIES PUBLIC_HEADER "${public_headers}")
 
+include(CMakePackageConfigHelpers)
+set(MATIO_CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}" CACHE
+    STRING "install path for xmatioConfig.cmake")
+
+configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/matioConfig.cmake.in
+                              "${CMAKE_CURRENT_BINARY_DIR}/matioConfig.cmake"
+                              INSTALL_DESTINATION ${MATIO_CMAKECONFIG_INSTALL_DIR})
+
 # 'make install' to the correct locations (provided by GNUInstallDirs).
-install(TARGETS matio matio-static EXPORT matio-config
-        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+install(TARGETS matio matio-static EXPORT matio-targets
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/matio
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(EXPORT matio-config NAMESPACE MATIO:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/matioConfig.cmake
```

Comment:
THis is for having a relocatable package for transitive dependency. `matio-config.cmake` is enough as long as we directly link with `matio`. However, if you want to package a library `intermediate` that depends on `matio`, you don't want to populate the `INTERFACE_INCLUDE_DIRECTORIES` of `intermediate` with the includes from `matio`. Indeed the paths (to `matio` include directories) would be absolute path, specific to the machine the package was made on.

To solve this problem, you need to access an IMPORTED target of `matio` that will populate the include directories (and also the matio_FOUND and matio_LIBRARY) so it can be used by target linking with `intermediate`. That is the purpose of  `matioConfig.cmake.in`